plugins/woocommerce/includes/class-wc-install.php

```diff
@@ -1163,6 +1167,133 @@ public static function create_terms() {
 		}
 	}
 
+	/**
+	 * Install and activate the WooCommerce Legacy REST API plugin from the WordPress.org directory if all the following is true:
+	 *
+	 * 1. We are in a WooCommerce upgrade process (not a new install).
+	 * 2. The 'woocommerce_skip_legacy_rest_api_plugin_auto_install' filter returns false (which is the default).
+	 * 3. The plugin is not installed and active already (but see note about multisite below).
+	 * 4. The Legacy REST API is enabled in the site OR the site has at least one webhook defined that uses the Legacy REST API payload format (disabled webhooks also count).
```

Comment:
Might be worth noting here the additional condition that the legacy API must be in use.

Reply:
It's stated in "4. The Legacy REST API is enabled in the site". If you mean checking _actual_ usages by looking at the information logged (https://github.com/woocommerce/woocommerce/pull/41804), I think it's better to play safe and just consider that API enabled === API used.